tests/swo_aws_extension/test_shared.py

```diff
@@ -0,0 +1,15 @@
+from decimal import Decimal
+
+from swo_aws_extension.shared import find_first_match
+
+
+def test_find_first_match_found():
+    d = {"a": Decimal("1.0"), "b": Decimal("2.0")}
```

Comment:
Ummm, it's really strange that the check doesn't fail here. Are we sure the checks are properly configured in this repo?

In addition, same comment about the AAA pattern

````python
    decimal = {"a": Decimal("1.0"), "b": Decimal("2.0")}
    
    result = find_first_match(decimal, "a", Decimal("1.0"))

    assert result == {"a": Decimal("1.0")}
````